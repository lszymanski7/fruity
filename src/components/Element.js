import React from 'react'
import PropTypes from 'prop-types'

const Element = ({ checked, handleCheck, id, img, name }) => {
    return (
        <div className="element">
            <label className="element__label">
                <input
                    checked={checked}
                    className="element__checkbox"
                    onChange={() => handleCheck(id)}
                    type="checkbox"
                />
                <img
                    alt={name}
                    className="element__image"
                    src={img}
                />
                <span className="element__caption">{name.toUpperCase()}</span>
            </label>
        </div>
    )
}

Element.propTypes = {
    checked: PropTypes.bool.isRequired,
    handleCheck: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}

export { Element as default }
