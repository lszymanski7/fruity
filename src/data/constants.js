const ICONS = {
    apple: require('../assets/images/icons/apple.webp'),
    banana: require('../assets/images/icons/banana.webp'),
    blackberry: require('../assets/images/icons/blackberry.webp'),
    blueberries: require('../assets/images/icons/blueberries.webp'),
    cherries: require('../assets/images/icons/cherries.webp'),
    coconut: require('../assets/images/icons/coconut.webp'),
    dragon_fruit: require('../assets/images/icons/dragon-fruit.webp'),
    fig: require('../assets/images/icons/fig.webp'),
    grapes: require('../assets/images/icons/grapes.webp'),
    kiwi: require('../assets/images/icons/kiwi.webp'),
    lemon: require('../assets/images/icons/lemon.webp'),
    lychee: require('../assets/images/icons/lychee.webp'),
    mango: require('../assets/images/icons/mango.webp'),
    melon: require('../assets/images/icons/melon.webp'),
    orange: require('../assets/images/icons/orange.webp'),
    papaya: require('../assets/images/icons/papaya.webp'),
    peach: require('../assets/images/icons/peach.webp'),
    pear: require('../assets/images/icons/pear.webp'),
    pineapple: require('../assets/images/icons/pineapple.webp'),
    plum: require('../assets/images/icons/plum.webp'),
    pomegranate: require('../assets/images/icons/pomegranate.webp'),
    star_fruit: require('../assets/images/icons/star-fruit.webp'),
    strawberry: require('../assets/images/icons/strawberry.webp'),
    watermelon: require('../assets/images/icons/watermelon.webp')
}

const LOGOS = {
    fruity: {
        size_128x128: require('../assets/images/logos/fruity-128x128.webp'),
        size_256x256: require('../assets/images/logos/fruity-256x256.webp'),
        size_512x512: require('../assets/images/logos/fruity-512x512.webp')
    },
    github: require('../assets/images/logos/github.svg')
}

const SHAPES = {
    moon: require('../assets/images/shapes/moon.svg'),
    reset: require('../assets/images/shapes/reset.svg'),
    sun: require('../assets/images/shapes/sun.svg')
}

export { ICONS, LOGOS, SHAPES }
