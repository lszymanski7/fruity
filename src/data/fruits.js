import { ICONS } from './constants'

// Fruits data
const fruits = [
    {
        id: 1,
        name: 'apple',
        img: ICONS.apple,
        checked: false
    },
    {
        id: 2,
        name: 'banana',
        img: ICONS.banana,
        checked: false
    },
    {
        id: 3,
        name: 'blackberry',
        img: ICONS.blackberry,
        checked: false
    },
    {
        id: 4,
        name: 'blueberries',
        img: ICONS.blueberries,
        checked: false
    },
    {
        id: 5,
        name: 'cherries',
        img: ICONS.cherries,
        checked: false
    },
    {
        id: 6,
        name: 'coconut',
        img: ICONS.coconut,
        checked: false
    },
    {
        id: 7,
        name: 'dragon fruit',
        img: ICONS.dragon_fruit,
        checked: false
    },
    {
        id: 8,
        name: 'fig',
        img: ICONS.fig,
        checked: false
    },
    {
        id: 9,
        name: 'grapes',
        img: ICONS.grapes,
        checked: false
    },
    {
        id: 10,
        name: 'kiwi',
        img: ICONS.kiwi,
        checked: false
    },
    {
        id: 11,
        name: 'lemon',
        img: ICONS.lemon,
        checked: false
    },
    {
        id: 12,
        name: 'lychee',
        img: ICONS.lychee,
        checked: false
    },
    {
        id: 13,
        name: 'mango',
        img: ICONS.mango,
        checked: false
    },
    {
        id: 14,
        name: 'melon',
        img: ICONS.melon,
        checked: false
    },
    {
        id: 15,
        name: 'orange',
        img: ICONS.orange,
        checked: false
    },
    {
        id: 16,
        name: 'papaya',
        img: ICONS.papaya,
        checked: false
    },
    {
        id: 17,
        name: 'peach',
        img: ICONS.peach,
        checked: false
    },
    {
        id: 18,
        name: 'pear',
        img: ICONS.pear,
        checked: false
    },
    {
        id: 19,
        name: 'pineapple',
        img: ICONS.pineapple,
        checked: false
    },
    {
        id: 20,
        name: 'plum',
        img: ICONS.plum,
        checked: false
    },
    {
        id: 21,
        name: 'pomegranate',
        img: ICONS.pomegranate,
        checked: false
    },
    {
        id: 22,
        name: 'star fruit',
        img: ICONS.star_fruit,
        checked: false
    },
    {
        id: 23,
        name: 'strawberry',
        img: ICONS.strawberry,
        checked: false
    },
    {
        id: 24,
        name: 'watermelon',
        img: ICONS.watermelon,
        checked: false
    }
]

export { fruits as default }
