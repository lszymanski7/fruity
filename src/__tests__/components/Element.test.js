import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Element from '../../components/Element'
import { fruits } from '../../data/constants'

const props = {
    checked: fruits[0].checked,
    handleCheck: jest.fn(),
    id: fruits[0].id,
    img: fruits[0].img,
    name: fruits[0].name
}

describe('Element', () => {
	it('should be rendered correctly.', () => {
		const { container } = render(<Element {...props} />)
		expect(container).toMatchSnapshot()
	})

    it('should handle clicking.', async () => {
        render(<Element {...props} />)
        const checkbox = screen.getByRole('checkbox')
        await userEvent.click(checkbox)
        expect(props.handleCheck).toHaveBeenCalled()
    })

	it('should have the correct image.', () => {
		render(<Element {...props} />)
		const img = screen.getByRole('img')
        expect(img).toHaveAttribute('alt', props.name)
		expect(img).toHaveAttribute('src', props.img)
	})

    it('should have the correct title.', () => {
		render(<Element {...props} />)
		const title = props.name
		const h3 = screen.getByRole('heading', { level: 3 })
		expect(h3).toHaveTextContent(title)
	})
})
