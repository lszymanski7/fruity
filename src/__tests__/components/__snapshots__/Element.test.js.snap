// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element should be rendered correctly. 1`] = `
<div>
  <div
    class="item"
  >
    <input
      id="bbb33b2d-e07d-4d13-a869-f3790e100477"
      type="checkbox"
    />
    <label
      for="bbb33b2d-e07d-4d13-a869-f3790e100477"
    >
      <img
        alt="Apple"
        class="item__image"
        src="apple.webp"
      />
    </label>
    <h3
      class="item__title"
    >
      Apple
    </h3>
  </div>
</div>
`;
